src/material/table/testing/row-harness.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate} from '@angular/cdk/testing';
+import {RowHarnessFilters, CellHarnessFilters} from './table-harness-filters';
+import {MatCellHarness, MatHeaderCellHarness, MatFooterCellHarness} from './cell-harness';
+
+/** Harness for interacting with a standard Angular Material table row. */
+export class MatRowHarness extends ComponentHarness {
+  static hostSelector = '.mat-row';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a row with specific attributes.
+   */
+  static with(options: RowHarnessFilters = {}): HarnessPredicate<MatRowHarness> {
+    return new HarnessPredicate(MatRowHarness, options);
+  }
+
+  /** Gets all cells of the table row. */
+  async getCells(filter: CellHarnessFilters = {}): Promise<MatCellHarness[]> {
+    return this.locatorForAll(MatCellHarness.with(filter))();
+  }
+
+  /** Gets the text of the cells in the row. */
+  async getCellTextByIndex(filter: CellHarnessFilters = {}): Promise<string[]> {
+    return getCellTextByIndex(this, filter);
+  }
```

Comment:
I think we decided to add `getCellTextByColumnName` to the row as well, but that could always be added in a follow-up

Reply:
Added to my list to do in a follow-up PR.